core/templates/hash_map.h

```diff
@@ -541,60 +538,73 @@ class HashMap {
 
 	TValue &operator[](const TKey &p_key) {
 		uint32_t pos = 0;
-		bool exists = _lookup_pos(p_key, pos);
-		if (!exists) {
-			return _insert(p_key, TValue())->data.value;
-		} else {
+		const uint32_t hash = _hash(p_key);
+		bool exists = _lookup_pos_with_hash(p_key, pos, hash);
+		if (exists) {
 			return elements[pos]->data.value;
+		} else {
+			return _insert_with_hash(p_key, TValue(), hash)->data.value;
 		}
 	}
 
 	/* Insert */
 
 	Iterator insert(const TKey &p_key, const TValue &p_value, bool p_front_insert = false) {
-		return Iterator(_insert(p_key, p_value, p_front_insert));
+		uint32_t pos = 0;
+		const uint32_t hash = _hash(p_key);
+		bool exists = _lookup_pos_with_hash(p_key, pos, hash);
+		if (!exists) {
+			return Iterator(_insert_with_hash(p_key, p_value, hash, p_front_insert));
+		} else {
+			elements[pos]->data.value = p_value;
+			return Iterator(elements[pos]);
+		}
```

Comment:
```suggestion
		if (exists) {
			elements[pos]->data.value = p_value;
			return Iterator(elements[pos]);
		} else {
			return Iterator(_insert_with_hash(p_key, p_value, hash, p_front_insert));
		}
```
I'd flip this to make it more readable

Reply:
I did this because when using the `insert` method we are more likely to want to insert a new element rather than modify an existing one. But using unlikely is a bit risky.